active-rfcs/0000-vtu-api.md

```diff
@@ -0,0 +1,268 @@
+- Simple plugin system to allow extending VTU with your own methods. See [Plugins](#plugin-system)
+
+## API changes
+
+We will only list the changes and deprecations. Please check the temporary [documentation](https://vuejs.github.io/vue-test-utils-next-docs/) for a full API listing.
+
+### mountOptions
+
+#### props
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#props) - Renamed from `propsData` to match component `props` field.
+
+#### global
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#global) - The `global` namespace is used to pass configuration to the `createApp` instance. Things like global components, directives and so on.
+
+These settings can also be globally set via the exported `config` object - `config.global.mocks.$t = jest.fn()`.
+
+- **global.components** - register global components
+- **global.directives** - register a global directive
+- **global.mixins** - register a global mixin
+- **global.plugins** - install a plugin
+- **global.stubs** - see bellow
+- **global.mocks** - see bellow
+- **global.provide** - see bellow
+
+#### stubs 
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#global-stubs) - Moved to `global.stubs`. 
+
+- **New** - Stubs will no longer render the slots of a component. This can be enabled by a global flag `config.renderStubSlots = true`.
+
+#### mocks
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#global-mocks) - Moved to `global.mocks`
+
+#### provide
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#global-provide) - Moved to `global.provide`.
+
+### Methods
+
+#### classes
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#classes)
+
+- **New** - throw error for multiple root nodes.
+
+#### unmount
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#unmount)
+
+- **New** replaces `destroy` to match Vue 3 API
+
+#### find
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#find)
+
+- **Breaking** - Returns only `DOMWrapper`. Cannot find Component instances. see [findComponent](#findcomponent)
+- **Breaking** - Accepts query selector only. 
+- **New** - Can now return instance root element, or a fragment of the root.
+
+#### findAll
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#findall)
+
+- **Breaking** - Returns only array of `DOMWrapper`. 
+- **Breaking** - No longer returns `WrapperArray`.
+- **Breaking** - Accepts query selector only. 
+
+#### findComponent
+
+[Link](https://vuejs.github.io/vue-test-utils-next-docs/api/#findcomponent)
+
+`findComponent` can find component instances, deeply nested in your component tree. This is most useful for edge case assertions, that are not reflected in the DOM or when using `shallowMount` and asserting props on a stub.
+ 
+Users should use `find` for the majority of cases, asserting the DOM properties, not the Vue components.
```

Comment:
Without proper explanation this line definitely sounds confusing and opinionated. What benefits will end user loose (from VTU part, not from "testing philosophy") if he/she chooses to go with `findComponent` approach? 

It would be really great if library specifically pick which end of an egg a person cracks to explain downsides for library consumers

 Is this line an official Vue.js team recommendation? How this aligns with Vue3 being split into "core" and "renderer" parts, so not all Vue components might

Reply:
Also correct. Chosing how to test is up to the developer. We decided we cannot enforce just one way of querying.

'findComponent' finds only Vue Components and cannot find plain dom elements. We hope this allows test suites running shallowMount to still be able to assert child Components, while making the tests read a bit cleaner, plus the result type be more predictable.

Will reword this a bit, thank you.

Reply:
The points you are raising are very likely to be addressed in the docs. From my understanding, this sentence is an argument for the sentence right before